pkg/build/generator/generator.go

```diff
@@ -515,8 +513,8 @@ func generateBuildFromBuild(build *buildapi.Build) *buildapi.Build {
 	return &buildapi.Build{
 		Spec: buildCopy.Spec,
 		ObjectMeta: kapi.ObjectMeta{
-			Name:   getNextBuildNameFromBuild(buildCopy),
-			Labels: buildCopy.ObjectMeta.Labels,
+			GenerateName: buildCopy.Name + "-",
```

Comment:
if you clone a clone, this will be the result, right?
```
build-1-asldkaj
build-1-asldkaj-qwueqw
build-1-asldkaj-qwueqw-xsasld
```

what would the old method do with a clone of a clone?